api/database/migrations/2024_05_27_202609_user_search_index_table.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('user_search_indices', function (Blueprint $table) {
+            $table->uuid('id')->primary('id')->default(new Expression('gen_random_uuid()'));
+            $table->foreign('id')->references('id')->on('users')->onUpdate('cascade')->onDelete('cascade');
+        });
+
+        DB::statement('DROP INDEX users_searchable_index');
+        DB::statement('ALTER TABLE user_search_indices ADD COLUMN searchable TSVECTOR');
+        DB::statement('CREATE INDEX users_searchable_index ON user_search_indices USING GIN(searchable)');
+
+        // copy existing values before dropping column
+        $users = DB::table('users')->select(['id', 'searchable'])->get();
```

Comment:
Try to avoid using looping to work with database operations.  It's slow and since it's not atomic you might miss rows if there's databse activity while it's running.  Try to do something like 
```sql
insert into user_search_indices (id, searchable)
select id, searchable from users
```

Reply:
Okay, seems reasonable to me. Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10516/commits/604bfbe9b22c7b399995428eef7ccef0065141d9

Reply:
Looks great!